client/scss/components/forms/_field.scss

```diff
@@ -61,6 +61,11 @@
   padding-top: theme('spacing.[2.5]');
   // This padding gives room for widgets such as comments beside of inputs
   padding-inline-end: calc($comment-button-size + $comment-button-space);
+
+  .w-field__input {
+    // No need for extra padding for nested field inputs.
+    padding-inline-end: 0;
+  }
```

Comment:
This isn’t super related to the rest here, but thought this would be a quick win.